core/src/main/java/org/apache/accumulo/core/spi/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -330,6 +334,8 @@ public Supplier<Map<String,String>> load(TableId key) {
           }
         });
 
+    migrationLimitExceededWarningCache = Caffeine.newBuilder().expireAfterWrite(5, MINUTES).build();
```

Comment:
Please see `EscalatingLogger` in https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/logging/ConditionalLogger.java. It's used at https://github.com/apache/accumulo/blob/main/server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java#L123. I think this can be re-used here.

Reply:
Oh that's way better! 
I'll switch this code over to using that.